app/src/main/java/com/example/sunasterisk/MainActivity.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.sunasterisk
+
+import com.example.android_template.R
+import com.example.sunasterisk.base.BaseActivity
+import com.example.sunasterisk.base.BasePresenter
+
+class MainActivity : BaseActivity<BasePresenter<*>>() {
```

Comment:
```suggestion
class MainActivity : BaseActivity() {
```
Activity & Fragment ở đây là View,
Mình phải tách BasePresenter ra khỏi View.

Reply:
Dạ, đoạn này em mong muốn generic type khi khai báo presenter trong base activity ạ anh.  Cái main activity này em quên chưa chỉnh lại ạ,